### DISCUSSION TITLE
Check ID / Use Only Host ID

### DISCUSSION THREAD

**Discussion by xBr0th3rx** [rustdesk/rustdesk-server-pro#482] [General]
We have Rustdesk Server Pro in the company and only use it for Debian and Ubuntu.

But now we have the problem that we clone some systems for operational reasons and this Rustdesk ID is then automatically taken over. The ID remains set until I run the service stop 
rm -r “~.config/rustdesk” 
rm -r “/root/.config/rustdesk”
Service start

This first generates a new ID.
Is it possible to add a check if the ID is already in use that it generates a new one?

What I am also wondering is what is the purpose of saving the data in ~/.config and /root/config?

Because I noticed that if a user does not have a rustdesk folder he copies it from the root directory and vice versa.

Because we have synced homes and the users do not always work in the same place which leads to several ID's being generated.

**Comment by rustdesk** [comment]
> Is it possible to add a check if the ID is already in use that it generates a new one?

Can not yet

**Reply by xBr0th3rx** [reply]
Is it possible to add this function?

Or whether you can set in the client that it automatically registers with the ID of the hostname? because we always have unique hostnames so there shouldn't be any duplicate IDs.

**Reply by rustdesk** [reply]
Sorry, we can not.

**Reply by rustdesk** [reply]
> unique hostnames

You can ask for a feature request for this. It does make some sense to me so far, we may consider to add it (add to custom client advanced options).

https://github.com/rustdesk/rustdesk/wiki/FAQ#feature-request-policy

**Reply by rustdesk** [reply]
> You can ask for a feature request for this. It does make some sense to me so far, we may consider to add it (add to custom client advanced options).

But this can not help if you clone rustdesk id with config files to other machines. It only works for newly installed rustdesk.

**Reply by xBr0th3rx** [reply]
Are there other ways to check that a client with the same ID is trying to connect in such cases?

I have already thought about only starting rustdesk for the clones when they have their first start, but currently we already have a few in circulation.

**Reply by rustdesk** [reply]
rm -r “~.config/rustdesk”
rm -r “/root/.config/rustdesk”

then restart, might be the easiest way.

**Reply by rustdesk** [reply]
Or you reset machine id ( /etc/machine-id) of each machine, then restart rustdesk. But be carefully, I do not know its bad effect.

**Reply by xBr0th3rx** [reply]
> rm -r “~.config/rustdesk” rm -r “/root/.config/rustdesk”
> 
> then restart, might be the easiest way.

Yes, I thought about that for the script.

**Reply by xBr0th3rx** [reply]
Could you explain to me exactly how the process works?

When the computer boots up and rustdesk is installed, it creates a config in /root/.config/rustdesk.

When a user logs on, the config is copied from root to user, right?

But what if a user logs in but has a different config because he was working on a different computer before?

**Reply by xBr0th3rx** [reply]
Is the config still overwritten from root to user or is the config overwritten from user to root?

**Reply by rustdesk** [reply]
Both, there is a sync mechanism.

**Reply by xBr0th3rx** [reply]
Okay, but that poses a problem for us.

Because we have shared workstations and synchronized home profiles that do not remain on the clients after shutdown.

Scenario:
Computer 1 is installed rustdesk and receives an id.
User1 logs on and does not yet have a rustdesk config which is copied from root when logging on.
User logs off and the computer shuts down.

Computer 2 rustdesk is installed and User1 logs in the config is copied from User1 to root. This means that computer 2 has the same config or id as computer 1.

A new user2 now logs on to computer 1 who does not yet have a rustdesk config and then receives the config from root. the user therefore has the same config as computer 2 

This would result in simultaneous IDs or not?

**Reply by rustdesk** [reply]
RustDesk's design is all users on the same machine having the same ID. RustDesk may not fit you if you require different user on the same machine having different id.

**Reply by xBr0th3rx** [reply]
Okay that means it always takes the config from root right?

**Comment by rustdesk** [comment]
> What I am also wondering is what is the purpose of saving the data in ~/.config and /root/config?

This is for unattended access, e.g. when you reboot your machine and enter login screen, there is only a `root` user.

**Reply by xBr0th3rx** [reply]
Ah Okay, I see, otherwise you wouldn't be able to log in to the login screen in the first place, right?

**Reply by rustdesk** [reply]
Yes
